src/sys/time.rs

```diff
@@ -58,6 +60,20 @@ const TS_MAX_SECONDS: i64 = ::std::isize::MAX as i64;
 
 const TS_MIN_SECONDS: i64 = -TS_MAX_SECONDS;
 
+impl From<Duration> for TimeSpec {
+    fn from(duration: Duration) -> Self {
+        TimeSpec(timespec{
+            tv_sec: cmp::min(duration.as_secs(), time_t::max_value() as u64) as time_t,
```

Comment:
I'm a little wary of this conversion. How do we document this and also how do we report an error to the user? I don't like just capping this value, I'd actually suggest a `panic!` instead.

Reply:
I don't think the conversion really matter since it's silently capped to 24 hours by the OS in any case. But I see it panic on i386, so I'll have to fix that anyway.

Reply:
That of course only applied to `kevent`, I'll fix this as well.